components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/PostAuthnMissingClaimHandler.java

```diff
@@ -312,6 +320,15 @@ protected void handlePostAuthenticationForMissingClaimsResponse(HttpServletReque
             } catch (UserStoreException e) {
                 if (e instanceof UserStoreClientException) {
                     context.setProperty(POST_AUTH_MISSING_CLAIMS_ERROR, e.getMessage());
+                    /*
+                    When the attribute update is disabled for JIT provisioned users, the manadatory claim update
+                    request will be identified through the error code and handled it.
+                     */
+                    if (ERROR_CODE_INVALID_ATTRIBUTE_UPDATE.equals(e.getErrorCode())) {
+                        context.getSequenceConfig().getAuthenticatedUser().
```

Comment:
Let's not throw PostAuthenticationFailedException in this case.

Reply:
Fixed https://github.com/wso2/carbon-identity-framework/pull/3641/commits/8c983fa31f698945cedfebf5e7c9f534d0a3ab3a